### PR TITLE
WSTEAM1-1068: Add csp next

### PR DESCRIPTION
**Pull request by shayneahchoon**
Resolves JIRA [WSTEAM1-1068](https://jira.dev.bbc.co.uk/browse/WSTEAM1-1068)

Overall changes
======
Add the following csp headers onto the Next.js app:
```
 policiesFromExpress= [
  'default-src',
  'child-src',
  'connect-src',
  'font-src',
  'frame-src',
  'img-src',
  'script-src',
  'style-src',
  'media-src',
  'worker-src',
  'report-to',
  'upgrade-insecure-requests',
];
```

Code changes
======
- _Next.js middleware function altered to append csp onto both the request and response headers._
- _Helmet csp function in the express app refactored to prevent dependency resolution errors from Next.js app._

Testing
======
1. _List the steps used to test this PR._

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: ws-nextjs-app/middleware.page.ts [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
       return NextResponse.rewrite(new URL('/sw.js', request.url));
     }
+  } else if (process.env.NODE_ENV === 'production') {
+    return cspHeaderResponse({
+      request,
+      service,
+      isAmp,
+      isLive: isLiveEnv(),
+      reportOnlyOnLive: service === 'japanese',
+    });
```
Comment (amoore108) [inline ws-nextjs-app/middleware.page.ts]: Probably could just pass in the `request` into here and then figure out the other parameters within `cspHeaderResponse` from that. Things like `isLive` can be imported directly in that `cspHeaderResponse` function too, just to keep it all encapsulated.

File: ws-nextjs-app/utilities/cspHeaderResponse/index.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    reportOnlyOnLive,
+    service,
+  });
+
+  const nextJSSpecificHeaders = {
+    'base-uri': 'self',
+    'form-action': 'self',
+    'frame-ancestors': 'none',
+    'object-src': 'none',
+  };
```
Comment (amoore108) [inline ws-nextjs-app/utilities/cspHeaderResponse/index.ts]: Not sure we need these. Its worth just trying the existing directives we have in the Express app and seeing that that list works.

File: ws-nextjs-app/utilities/cspHeaderResponse/index.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  service: Services;
+}) => {
+  const requestHeaders = new Headers(request.headers);
+
+  const helmetCSP = helmetCsp({
+    isAmp,
+    isLive,
+    reportOnlyOnLive,
+    service,
+  });
```
Comment (amoore108) [inline ws-nextjs-app/utilities/cspHeaderResponse/index.ts]: Could this be renamed to something more generic than `helmetCsp`? HelmetCsp is an actual library, so just don't want to cause confusion that the Next.js app will actually be using that library.

File: src/server/utilities/cspHeader/directives.js [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+    'style-src': generateStyleSrc({ isAmp, isLive }),
+    'media-src':
+      service === 'mundo'
+        ? "'self' blob: https:"
+        : generateMediaSrc({ isAmp, isLive }),
+    'worker-src': generateWorkerSrc({ isAmp }),
+    'report-to': 'worldsvc',
+    'upgrade-insecure-requests': [],
+  },
+  reportOnly: reportOnlyOnLive,
```
Comment (amoore108) [inline src/server/utilities/cspHeader/directives.js]: Should remove the `reportOnlyOnLive` parameter and just check `service === 'japanese'` here? Would save having to update that logic in 2 places.

**Review by amoore108 — APPROVED** [review]
Looks good! I'd test this on the Preview environment first against some Live page assets that have embeds/video players on them just to be sure.

**Review by emilysaffron — APPROVED** [review]
🎉

**Pull request merged**